falcon/util/streams.py

```diff
@@ -0,0 +1,276 @@
+import functools
+import io
+
+DEFAULT_CHUNK_SIZE = io.DEFAULT_BUFFER_SIZE * 4
+
+
+class BufferedStream:
+
+    def __init__(self, read, max_stream_len, chunk_size=None):
+        self._read_func = read
```

Comment:
It may make sense here to accept a file-like "stream" to wrap here, rather than the read method itself, ala `io.BufferedReader()`.

Reply:
Nevermind; I see how this argument is used by `delimit()`.